File: src/main/java/com/employee/biodata/model/Employee.java
package com.employee.biodata.model;

import java.util.HashSet;
import java.util.Set;

import javax.persistence.*;

import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;

@Entity
@Table(name = "employee")
@JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class, property = "id")
public class Employee {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	
	private String firstName;

	private String middleName;

	private String lastName;

	private String email;

	private String username;

	private String password;

	private String phoneNumber;

	private String address;

	private double height;

	private double weight;
	
	@OneToMany(mappedBy = "employee", cascade = CascadeType.ALL)
	private Set<EmployeeDetails> employeeDetails = new HashSet<>();

	@OneToMany(cascade = CascadeType.ALL, fetch = FetchType.EAGER, mappedBy = "employee")
	private Set<Authority> authorities = new HashSet<>();

	public Employee() {
	}

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public String getFirstName() {
		return firstName;
	}

	public void setFirstName(String firstName) {
		this.firstName = firstName;
	}

	public String getMiddleName() {
		return middleName;
	}

	public void setMiddleName(String middleName) {
		this.middleName = middleName;
	}

	public String getLastName() {
		return lastName;
	}

	public void setLastName(String lastName) {
		this.lastName = lastName;
	}

	public String getEmail() {
		return email;
	}

	public void setEmail(String email) {
		this.email = email;
	}

	public String getUsername() {
		return username;
	}

	public void setUsername(String username) {
		this.username = username;
	}

	public String getPassword() {
		return password;
	}

	public void setPassword(String password) {
		this.password = password;
	}

	public String getPhoneNumber() {
		return phoneNumber;
	}

	public void setPhoneNumber(String phoneNumber) {
		this.phoneNumber = phoneNumber;
	}

	public String getAddress() {
		return address;
	}

	public void setAddress(String address) {
		this.address = address;
	}

	public double getHeight() {
		return height;
	}

	public void setHeight(double height) {
		this.height = height;
	}

	public double getWeight() {
		return weight;
	}

	public void setWeight(double weight) {
		this.weight = weight;
	}

	public Set<EmployeeDetails> getEmployeeDetails() {
		return employeeDetails;
	}

	public void setEmployeeDetails(Set<EmployeeDetails> employeeDetails) {
		this.employeeDetails = employeeDetails;
	}

	public Set<Authority> getAuthorities() {
		return authorities;
	}

	public void setAuthorities(Set<Authority> authorities) {
		this.authorities = authorities;
	}
}

File: src/main/resources/static/employee.js
$(document).ready(function () {
    // $('.table .editButton').on('click', function (event) {
    //     event.preventDefault();
    //     var href = $(this).attr('href');
    //     $.get(href, function (employee, status) {
    //         $('#id').val(employee.id);
    //         $('#firstName').val(employee.firstName);
    //         $('#middleName').val(employee.middleName);
    //         $('#lastName').val(employee.lastName);
    //         $('#email').val(employee.email);
    //         $('#username').val(employee.username);
    //         $('#phoneNumber').val(employee.phoneNumber);
    //         $('#address').val(employee.address);
    //         $('#height').val(employee.height);
    //         $('#weight').val(employee.weight);
    //     });
    //     $('#editModal').modal();
    // });
    
    $('.table .deleteButton').on('click', function (event) {
        event.preventDefault();
        var href = $(this).attr('href');
        $('#deleteModal #deleteId').attr('href', href)
        $('#deleteModal').modal();
    });
});